.github/workflows/build_packages.yaml

```diff
@@ -557,6 +583,109 @@ jobs:
         aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
         docker push public.ecr.aws/emqx/emqx:${version#v}
 
+  docker-push-multi-arch-manifest:
+    # note, we only run on amd64
+    if: >
+      (github.event_name == 'release' && !github.event.release.prerelease)
+      || (github.event.repository.owner != 'emqx' && startsWith(github.ref_name, 'ci/'))
+    needs:
+      - prepare
+      - docker
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        profile: # all editions for docker
+          - emqx-edge
+          - emqx
+          - emqx-enterprise
+        # NOTE: for docker, only support latest otp version, not a matrix
+        otp:
+          - 24.1.5-4 # update to latest
+          #
+        elixir:
+          - 1.13.2 # update to latest
+        arch:
+          - amd64
+          - arm64
+        build_elixir:
+          - no_elixir
+        build_machine:
+          - aws-arm64
+          - ubuntu-20.04
+        exclude:
+          - arch: arm64
+            build_machine: ubuntu-20.04
+          - arch: amd64
+            build_machine: aws-arm64
+        include:
+          - os: alpine3.14
+            profile: emqx
+            otp: 24.1.5-4
+            elixir: 1.13.2
+            arch: amd64
+            build_elixir: with_elixir
+            build_machine: ubuntu-20.04
+
+    steps:
+      - uses: actions/download-artifact@v2
+        if: matrix.arch == 'amd64'
+        with:
+          name: source
+          path: .
+
+      - name: unzip source code
+        if:  matrix.arch == 'amd64'
+        run: unzip -q source.zip
+
+      - uses: docker/login-action@v1
+        if: matrix.arch == 'amd64'
+        with:
+          username: ${{ secrets.DOCKER_HUB_USER }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+      - name: prepare for docker-action-parms
+        id: pre-meta
```

Comment:
Since this is the sole step that does not check for `arch == amd64` and it (apparently) has no side effects other than setting outputs, maybe this whole job could only run with `amd64`?

Or does this step by itself have some purpose?

Reply:
Yes it looks stupid.
I have tried to add `if: matrix.arch == 'amd64'` under job and it didn't work get syntax error. that is why I have to set it on each step.

Reply:
What I meant was if we could simply remove `arm64` from the matrix. :smiley_cat:

Reply:
I want to keep the matrix part identical as the one in job `docker`, I tried to use some references to the `docker` job matrix something like `${{ needs.docker.strategy.matrix }}` but github action does not support that.

Reply:
Aha, I see!  Indeed, once I tried to use YAML anchors to avoid repetition, but it does not support that either...
Maybe later we'd have to resort to [dynamic matrices](https://tomasvotruba.com/blog/2020/11/16/how-to-make-dynamic-matrix-in-github-actions/) or a composite job. :speak_no_evil: